### PR TITLE
Attempt to fix Go Triage builds

### PR DESCRIPTION
**Pull request by michaelkolber**
Attempt to instead have Cloud Build build triage to avoid build context issues
/cc BenTheElder spiffxp

### CODE REVIEW

**Comment by BenTheElder** [comment]
https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-test-infra-push-triage/1291108031739727872

File: triage/cloudbuild.yaml [hunk 1]
```diff
@@ -5,14 +5,22 @@ steps:
...
+  - name: golang:$_GO_VERSION
+    args: ['go', 'build', '-v', '-o=/workspace/builds/triage', './triage/main.go']
+    env:
+      - GOOS=linux
+      - GOARCH=amd64
+      - GO111MODULE=on
+      - GOPROXY=https://proxy.golang.org
+      - GOSUMDB=sum.golang.org
   - name: gcr.io/cloud-builders/docker
-    args: [ 'build', '-t', 'gcr.io/$PROJECT_ID/triage:go-$_GIT_TAG', '-f', 'triage/Dockerfile-go', '.' ]
```
Comment (BenTheElder) [inline triage/cloudbuild.yaml:10]: pretty sure this should be ./triage/Dockerfile-go? whoops :-)

**Comment by spiffxp** [comment]
ref: https://github.com/kubernetes/test-infra/issues/18726

**Pull request merged**